### PR TITLE
Order cart functionality added

### PR DESCRIPTION
**Pull request by jszymanski97**
Order entity, dto, repository, controller and service layers added OrderItem entity, dto, repository, controller and service layers added Liquibase changelogs for creating Order and orderItem tables
 implemented

### CODE REVIEW

File: src/main/java/mate/project/dto/PlaceOrderDto.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package mate.project.dto;
+
+import jakarta.validation.constraints.NotNull;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class PlaceOrderDto {
```
Comment (slade13) [inline src/main/java/mate/project/dto/PlaceOrderDto.java:9]: The `@NotNull` annotation is used to ensure that the shippingAddress field is not null. However, it does not check for empty strings. Consider using `@NotBlank` from jakarta.validation.constraints if you want to ensure the field is not null and not empty.

File: src/main/java/mate/project/service/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -92,9 +92,8 @@ private User getAuthenticatedUser() {
     }
 
     private ShoppingCart getUsersShoppingCart(User user) {
-        return shoppingCartRepository.findByUser(user)
+        return shoppingCartRepository.findByUserWithCartItems(user)
```
Comment (slade13) [inline src/main/java/mate/project/service/ShoppingCartServiceImpl.java]: The method name `findByUserWithCartItems` suggests that it retrieves a user along with their cart items. Ensure that this method is implemented correctly in the repository and that it returns the expected data structure. If this method is new, verify its functionality with appropriate tests.

File: src/main/java/mate/project/service/OrderService.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import mate.project.dto.OrderResponseDto;
+import mate.project.dto.PlaceOrderDto;
+import mate.project.dto.StatusDto;
+
+public interface OrderService {
+    OrderResponseDto placeOrder(PlaceOrderDto placeOrderDto);
+
+    List<OrderResponseDto> findAll();
+
+    void update(Long id, StatusDto statusDto);
```
Comment (slade13) [inline src/main/java/mate/project/service/OrderService.java]: The update method does not return any value. Consider returning a boolean or the updated object to indicate success or failure of the update operation.

File: src/main/java/mate/project/service/OrderService.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+    List<OrderItemResponseDto> findOrderItemsByOrderId(Long orderId);
+
+    OrderItemResponseDto getOrderItemByIds(Long orderId, Long itemId);
```
Comment (slade13) [inline src/main/java/mate/project/service/OrderService.java]: The method name 'getOrderItemByIds' could be more descriptive. Consider renaming it to 'findOrderItemByOrderIdAndItemId' for better clarity.

File: src/main/resources/db/changelog/changes/12-create-orders-table.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+              - column:
+                  name: order_date
+                  type: timestamp
+                  constraints:
+                    nullable: false
+              - column:
+                  name: shipping_address
+                  type: varchar(255)
+                  constraints:
+                    nullable: true
```
Comment (slade13) [inline src/main/resources/db/changelog/changes/12-create-orders-table.yaml:42]: Ensure shipping_address is consistent across the schema `nullable: true` and the DTO `@NotNull`.

**Review by slade13 — APPROVED** [review]
Good job, please review my comments!

File: src/main/java/mate/project/service/ShoppingCartServiceImpl.java (hunk 2 above)
Comment (jszymanski97) [reply src/main/java/mate/project/service/ShoppingCartServiceImpl.java]: Bad naming... It was a custom method returning just the ShoppingCart items for the speciffic user. Fixed

**Pull request merged**